contribs/emissions/src/main/java/org/matsim/contrib/emissions/WarmEmissionHandler.java

```diff
@@ -123,106 +135,110 @@ public void handleEvent(VehicleEntersTrafficEvent event) {
 			}
 		}
 		Tuple<Id<Link>, Double> linkId2Time = new Tuple<>(event.getLinkId(), event.getTime());
-		this.vehicleEntersTraffic.put(event.getVehicleId(), linkId2Time);
+		this.vehicleEntersTrafficMap.put(event.getVehicleId(), linkId2Time);
 	}
 
 	@Override
 	public void handleEvent(LinkEnterEvent event) {
 		Tuple<Id<Link>, Double> linkId2Time = new Tuple<>(event.getLinkId(), event.getTime());
-		this.linkenter.put(event.getVehicleId(), linkId2Time);
+		this.linkEnterMap.put(event.getVehicleId(), linkId2Time);
 	}
 
 	@Override
 	public void handleEvent(LinkLeaveEvent event) {
+
 		Id<Vehicle> vehicleId = event.getVehicleId();
 		Id<Link> linkId = event.getLinkId();
-		double leaveTime = event.getTime();
 		Link link = this.scenario.getNetwork().getLinks().get(linkId);
 		double linkLength = link.getLength();
+		double leaveTime = event.getTime();
 
-		if (linkLength == 0.) {
-			if (zeroLinkLengthWarnCnt == 0) {
-				logger.warn("Length of the link " + linkId + " is zero. No emissions will be estimated for this link. Make sure, this is intentional.");
-				logger.warn(Gbl.ONLYONCE);
-				zeroLinkLengthWarnCnt++;
-			}
-			return;
-		}
+		warnIfZeroLinkLength(linkId, linkLength);
 
 		// excluding links with zero lengths from leaveCnt. Amit July'17
```

Comment:
Is this comment still correct? zero length links seem to be counted as well.

Reply:
Nope, I had to add an if clause to fix this... thanks.